### PR TITLE
feat(wmg): preemptive filtering

### PR DESCRIPTION
**Pull request by atarashansky**
## Reason for Change

- #4280 
- By necessity, addresses part of #4314
- #4292 

## Changes

- Split the valid secondary filter dimensions logic into a separate endpoint `/filters`.
  - This was necessary to enable behavior where the endpoint returns _all_ filter values when organism is the only selected criteria. This behavior was not possible to implement in the `/query` endpoint without changing the `WmgQueryCriteria` requirements (i.e. at least one tissue and gene must be selected).
- Update the `useFilterDimensions` hook to query the `/filters` endpoint.
- Update the `GeneSearchBar` component to filter the tissues dropdown to include only the tissues present in the `tissue_terms` response from`/filters`.
- Remove all references to and dependencies on `selectedCellTypeIds`.
  - This is a vestigial code path to enable cell type deletion. It was causing a bug where clearing a secondary filter was not refreshing the Y axis to include all unfiltered cell types.
  - This includes removing the reset icon (replay.svg) and all associated dependencies.
  - This includes removing all references to the _concept_ of cell type deletion (e.g. `deleteGenesAndCellTypeIds` would be refactored to `deleteGenes`).
 - Enable the filter dropdowns regardless if tissues and genes are selected or not.

## Testing steps

- [RDEV STACK](https://atar-filters-frontend.rdev.single-cell.czi.technology/gene-expression)

## Notes for Reviewer
- Unit tests for the `/filters` endpoint have not been added yet.
  - These will be added next week as part of @atarashansky's on-call responsibilities. It makes the most sense to delay adding tests until next week when the WMG unit tests will be refactored and updated.

### CODE REVIEW

**Comment by tihuan** [comment]
@atarashansky Awesome work! Let's add e2e tests to document the new filter behaviors 🚀

**Comment by tihuan** [comment]
Also should we wait to merge this PR after Compare PR is merged? I think priority wise Compare is higher, so wanna make sure that feature gets merged first. What do people think?

CC: @signechambers1

**Comment by tihuan** [comment]
@atarashansky I haven't had time to troubleshoot this bug, but repro steps:

1. Select tissues: blood and lung
2. Select dataset filter: first option
3. Click on tissue
4. Crashed the app due to runtime error `Cannot read properties of undefined (reading 'toString')`

I checked the `/filters` response and it returns just blood, no lung anymore since it's filtered out. So I'm thinking it's probably because the Redux store state for `selectedTissues` is still blood and lung, but now the returned tissue is just blood, so there's now a discrepancy between API response tissues (just blood) vs Redux state `selectedTissues` (still blood and lung)

I think _probably_ we could resolve this by updating the reducer that when we get tissues response, we update `selectedTissues` to remove invalid selections from the state as well. Hope that makes sense and happy to help with that too 👍 

https://user-images.githubusercontent.com/6309723/224127499-4a747652-f808-4fe3-9883-f96b0739c352.mov

**Comment by tihuan** [comment]
A perf regression on `/query` abortion is not working at the moment

Repro:

1. Open network tab
2. Click a few genes
3. Old `/query` requests should be cancelled

https://user-images.githubusercontent.com/6309723/224168179-d05416c2-2b45-40b3-b17e-4285839493a0.mov

**Review by tihuan — CHANGES_REQUESTED** [review]
Code looks great! Just one tiny nit and the two bugs reported above 👍 Let me know how I can help! 

Thanks for adding this exciting feature, Alec 🚀 !

File: frontend/src/views/WheresMyGene/components/GeneSearchBar/components/QuickSelect/index.tsx
Comment (atarashansky) [reply frontend/src/views/WheresMyGene/components/GeneSearchBar/components/QuickSelect/index.tsx:257]: It wouldn't. It needs to be stateful so that you don't get any visual glitch in the dropdown (e.g. temporary clearing of options) while the client is loading.

**Comment by tihuan** [comment]
> A perf regression on `/query` abortion is not working at the moment
> 
> Repro:
> 
> 1. Open network tab
> 2. Click a few genes
> 3. Old `/query` requests should be cancelled

@ashin-czi can you try to repro this please? @atarashansky can't repro this on his machine 🤔 Thank you!

**Review by tihuan — APPROVED** [review]
LGTM 🚀 !!

**Pull request merged**
